index.js

```diff
@@ -106,10 +106,19 @@ module.exports = {
       'components/dialog/dialog-theme.scss'
     ];
 
-    var pathBase = this.project.nodeModulesPath;
-    var angularMaterialPath =  'angular-material-source/src';
-
-    var angularScssFiles = new Funnel(path.join(pathBase,angularMaterialPath), {
+    /*
+      Find the angular-material-source module in a manner that works for npm 2.x
+      and 3.x in both the addon itself and projects that depend on this addon
+
+      This is an edge case b/c angular-material-source does not have a main
+      module we can require.resolve through node itself and similarily ember-cli
+      does not have such a hack for the same reason.
+
+      tl;dr - We want the non built scss files, and b/c this dep is only provided via
+      bower, we use this hack. Please change it if you read this and know a better way.
```

Comment:
They build to their dist directory and publish that to bower only. Their package.json has no main property and the project has no index.js file so tl;dr - there is no way to require their library from another node module - which after looking at some other addons like liquid fire and the node docs, would be our first choice for finding the path for all npm + node versions cleanly via 'path.dirname(require.resolve("angular-material-source"))/src' 

This is like the rest of the angular microlibs - they chose to use bower only for deps that used by angular 

We could file a ticket to add an empty index.js file in their repo but not sure they'd go for it. 

Check the node docs for the 'module' module towards the bottom and then try to find any of the termination steps that will resolve module.main in their code base and you'll see what I mean. I found none

And correct were not using bower, we want the actual source scss, but we're having to hack our own way to find it, which is just weird for sure - although currently ember uses the same pattern (until it becomes addon)

(on my phone ATM so let me know if that made sense - wow using mobile is hard on github, can't even see what I just wrote)